neuraxle/pipeline.py

```diff
@@ -249,6 +320,20 @@ def _find_starting_step_index(self, data_container: DataContainer) -> int:
 
         return index_latest_checkpoint
 
+    def _load_pipeline_cache(self, pipeline: 'Pipeline', new_starting_step_index):
+        steps_before_checkpoint_changed = False
+        for index, (step_name, step) in enumerate(self[:new_starting_step_index + 1]):
+            bytecode_current_step = dis.dis(step)
+            bytecode_cached_step = dis.dis(pipeline.steps[index])
+            if bytecode_current_step != bytecode_cached_step:
+                steps_before_checkpoint_changed = True
+
+        if not steps_before_checkpoint_changed:
+            self.set_hyperparams(pipeline.get_hyperparams())
+            self.set_hyperparams_space(pipeline.get_hyperparams_space())
+            self.steps = pipeline.steps[:new_starting_step_index] + \
```

Comment:
Something like this perhaps? I don't know what is `...` thought. 
```
for i, s in enumerate(self.steps):
    if s.can_load(...): 
        self.steps[i] = s.load(...)
```

Reply:
it would create a lot of io. I think saving the whole pipeline only when a checkpoint is saved is the right way to go. It is simpler, and more performant.

Reply:
to be discussed in the next design